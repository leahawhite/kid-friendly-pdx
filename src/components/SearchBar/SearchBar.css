.searchbar {
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 30px 10px;
  background-color: #b3f0ff;
  margin: 0 auto;
}

.search-form, .filter-form {
  text-align: left;
}

.search-form input {
  float: left;
  display: block;
  margin-bottom: 10px;
  padding: 5px 8px;
  text-align: left;
  width: 90%;
  border: 3px solid #00b8e6;
  border-radius: 8px 0 0 8px;
  font-weight: bold;
  font-family: 'Source Sans Pro';
}

::placeholder { /* Chrome, Firefox, Opera, Safari 10.1+ */
  color: #444;;
  opacity: 1; /* Firefox */
}

:-ms-input-placeholder { /* Internet Explorer 10-11 */
  color: #444;;
}

::-ms-input-placeholder { /* Microsoft Edge */
  color: #444;;
}

.search-btn {
  font-weight: bold;
  font-family: "Source Sans Pro";
  color: white;
  float: left;
  width: 10%;
  padding: 5px;
  background-color: #00b8e6;
  border: 3px solid #00b8e6;
  border-left: none;
  -moz-border-radius: 8px;
  -webkit-border-radius: 8px;
  -o-border-radius: 8px;
  -ms-border-radius: 8px;
  border-radius: 8px;
  border-top-left-radius: 0px;
  border-bottom-left-radius: 0px;
}

.filters {
  border: none;
  text-align: left;
  margin: 0;
  padding: 0;
}

.form-label, .form-select {
  padding: 5px 8px;
  border: 3px solid #00b8e6;
  background-color: white;
  font-size: 16px;
}

.form-label {
  display: inline-block;
  text-align: left;
  border-right: none;
  line-height: normal;
  font-weight: bold;
}

.form-select {
  -webkit-appearance:none;
  border-left: none;
  color: #444;
  font-family: 'Source Sans Pro';
}

.filter-category {
  margin-bottom: 10px;
}

.filter-category label, .filter-neighborhood label {
  border-radius: 8px 0 0 8px;
  padding: 5px 8px;
}

.filter-category select, .filter-neighborhood select {
  border-radius: 0 8px 8px 0;
}

.filter-category label {
  width: 85px;
}

.filter-category select {
  -webkit-appearance:none;
  width: calc(100% - 85px);
  padding: 5px 8px;
}

.filter-neighborhood label {
  width: 125px;
}

.filter-neighborhood select {
  -webkit-appearance:none;
  width: calc(100% - 125px);
  padding: 5px 8px;
}

input, option, button {
  cursor: pointer;
  font-size: 16px;
}

@media only screen and (min-width: 950px) {
  .searchbar {
    width: 80%;
  }
}

@media only screen and (min-width: 1200px) {
  .searchbar {
    width: 70%;
  }
}

