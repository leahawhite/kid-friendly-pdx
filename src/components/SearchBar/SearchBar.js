import React, { Component } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import './SearchBar.css'

class SearchBar extends Component {
  
  render() {
    const { 
      searchTerm, 
      category, 
      neighborhood, 
      handleUpdateSearch, 
      handleUpdateCategory, 
      handleUpdateNeighborhood, 
      handleSubmit } = this.props
    return (
      <section className="searchbar">
        <form className="search-form" onSubmit={handleSubmit}>
          <label className="search-label" htmlFor="search"></label>
          <input 
            className="search-input" 
            id="search" 
            type="text" 
            placeholder="Search:  pizza, cafe, park..."
            value={searchTerm}
            aria-label="search"
            onChange={handleUpdateSearch}/>
          <button type="submit" className="search-btn" aria-label="search">
            <FontAwesomeIcon icon="search"/>
          </button>
          <fieldset className="filters">
            <div className="filter-category">
              <label className="form-label" htmlFor="category">Category:</label>
              <select 
                  className="form-select" 
                  type="text" 
                  id="category"
                  value={category}
                  aria-label="category"
                  onChange={handleUpdateCategory}>
                    <option value=""></option>
                    <option value="restaurant">restaurant</option>
                    <option value="attraction">attraction</option>
                    <option value="museum">museum</option>
                    <option value="park">park</option>
                    <option value="indoor play">indoor play</option>
                    <option value="toy store">toy store</option>
                </select>
              </div>
            <div className="filter-neighborhood">
              <label className="form-label" htmlFor="neighborhood">Neighborhood:</label>
              <select 
                className="form-select" 
                type="text" 
                id="neighborhood"
                value={neighborhood}
                aria-label="neighborhood"
                onChange={handleUpdateNeighborhood}>
                  <option value=""></option>
                  <option value="N">N</option>
                  <option value="NE">NE</option>
                  <option value="NW">NW</option>
                  <option value="Downtown">Downtown</option>
                  <option value="SW">SW</option>
                  <option value="SE">SE</option>
                  <option value="Beaverton">Beaverton</option>
                  <option value="Vancouver">Vancouver</option>
                  <option value="Clackamas">Clackamas</option>
                  <option value="Milwaukie">Milwaukie</option>
              </select>
            </div>
          </fieldset> 
        </form>
      </section>
    )
  }
} 

export default SearchBar