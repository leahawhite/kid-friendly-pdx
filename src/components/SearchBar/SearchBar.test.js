import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchBar from './SearchBar';

describe('MapContainer component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchBar />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(<SearchBar />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})