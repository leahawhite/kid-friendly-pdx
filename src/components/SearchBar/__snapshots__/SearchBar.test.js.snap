// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer component renders UI as expected 1`] = `
<section
  className="searchbar"
>
  <form
    className="search-form"
  >
    <label
      className="search-label"
      htmlFor="search"
    />
    <input
      className="search-input"
      id="search"
      placeholder="Search"
      type="text"
    />
    <button
      className="search-btn"
      type="submit"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="search"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <fieldset
      className="filters"
    >
      <div
        className="filter-category"
      >
        <label
          className="form-label"
          htmlFor="category"
        >
          Category:
        </label>
        <select
          className="form-select"
          id="category"
          type="text"
        >
          <option
            value=""
          />
          <option
            value="restaurant"
          >
            restaurant
          </option>
          <option
            value="attraction"
          >
            attraction
          </option>
          <option
            value="museum"
          >
            museum
          </option>
          <option
            value="park"
          >
            park
          </option>
          <option
            value="indoor play"
          >
            indoor play
          </option>
          <option
            value="toy store"
          >
            toy store
          </option>
        </select>
      </div>
      <div
        className="filter-neighborhood"
      >
        <label
          className="form-label"
          htmlFor="neighborhood"
        >
          Neighborhood:
        </label>
        <select
          className="form-select"
          id="neighborhood"
          type="text"
        >
          <option
            value=""
          />
          <option
            value="N"
          >
            N
          </option>
          <option
            value="NE"
          >
            NE
          </option>
          <option
            value="NW"
          >
            NW
          </option>
          <option
            value="Downtown"
          >
            Downtown
          </option>
          <option
            value="SW"
          >
            SW
          </option>
          <option
            value="SE"
          >
            SE
          </option>
          <option
            value="Beaverton"
          >
            Beaverton
          </option>
          <option
            value="Vancouver"
          >
            Vancouver
          </option>
          <option
            value="Clackamas"
          >
            Clackamas
          </option>
          <option
            value="Milwaukie"
          >
            Milwaukie
          </option>
        </select>
      </div>
    </fieldset>
  </form>
</section>
`;
